api/napi/src/interpreter/component_instance.rs

```diff
@@ -118,6 +167,61 @@ impl JsComponentInstance {
         Err(napi::Error::from_reason(format!("{} is not a callback", callback_name).as_str()))
     }
 
+    #[napi]
+    pub fn set_global_callback(
+        &self,
+        env: Env,
+        global_name: String,
+        callback_name: String,
+        callback: JsFunction,
+    ) -> Result<()> {
+        let function_ref = RefCountedReference::new(&env, callback)?;
+
+        let ty = self
+            .inner
+            .definition()
+            .global_properties_and_callbacks(global_name.as_str())
+            .ok_or(napi::Error::from_reason(format!("Global {global_name} not found")))?
+            .find_map(|(name, proptype)| if name == callback_name { Some(proptype) } else { None })
+            .ok_or(())
+            .map_err(|_| {
+                napi::Error::from_reason(format!(
+                    "Callback {callback_name} of global {global_name} not found in the component"
```

Comment:
Please add a test that covers this code path (you can fold that into the existing test).